src/Symfony/Component/HttpKernel/Log/Recorder.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Log;
+
+use Psr\Log\LogLevel;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class Recorder implements DebugLoggerInterface
```

Comment:
I think we'd better merge that class with Logger and replace the setRecorder by eg `setDebug(bool $debug = true)`.

Reply:
How should I inject `RequestStack` in that case?

Reply:
`enableDebug(RequestStack|false)`?

Reply:
Would it be a problem if the logger depended on the request stack directly? 🤔

Reply:
To not break BC `RequestStack` should be made optional (because all constructor’s arguments are) but it would be required in “debug” mode so it seems weird to me.

`enableDebug(RequestStack|false)` sounds better to me though, can I implement it?

Reply:
maybe a decorating DebugLogger?

Reply:
A decorator might be a good idea. @MatTheCat WDYT?

Reply:
Can I decorate a service at runtime? I copied some implementation from the monolog and it's using a configurator 🤔

Reply:
you can't, but you don't have to since the debug flag is known at compile time
the DI extension could remove the decorating service when debug is off

Reply:
IIRC it was about the SAPI: the recorder was disabled when using the CLI.

Reply:
> the recorder was disabled when using the CLI

I don't see anything similar for DebugProcessor in the bridge, so why would we need that here?

Reply:
:facepalm: now I do, we call removeDebugLogger() in AddDebugLogProcessorPass!
Then let's reuse that method name, no need for a new one on LoggerPass.
And let's give the request stack as constructor argument yes.

Reply:
Hm I guess I must *append* `RequestStack` to the constructor’s arguments (to keep BC) but doing so implies I must set them all in the definition. Is that alright or would `enableDebug(RequestStack|false)` be better?

Reply:
setting all args look fine to me (but please don't used named arguments, only positional ones)